numpy/core/defchararray.py

```diff
@@ -51,21 +51,45 @@ def _use_unicode(*args):
     Helper function for determining the output type of some string
     operations.
 
-    For an operation on two ndarrays, if at least one is unicode, the
-    result should be unicode.
+    For an operation on two ndarrays, if at least one is a dtype with a
+    scalar that subclasses str or bytes, the result should be that
+    dtype. Otherwise, if at least one is unicode, the result should be
+    unicode.
     """
     for x in args:
-        if (isinstance(x, str) or
-                issubclass(numpy.asarray(x).dtype.type, unicode_)):
-            return unicode_
-    return string_
+        if isinstance(x, str):
+            return numpy.dtype(unicode_)
+        scalar_type = numpy.asarray(x).dtype.type
+        if isinstance(scalar_type, unicode_):
```

Comment:
```suggestion
        if issubclass(scalar_type, unicode_):
```
was the original code here.  I doubt it matters (the next path is probably just as well, and maybe better anyway).